src/pkg/bundle/create.go

```diff
@@ -86,16 +85,54 @@ func (b *Bundle) Create() error {
 		}
 	}
 
+	// for dev mode update package ref for local bundles, refs for remote bundles updated on deploy
+	if config.Dev && len(b.cfg.DevDeployOpts.Ref) != 0 {
+		for i, pkg := range b.bundle.Packages {
+			pkg = b.setPackageRef(pkg)
+			b.bundle.Packages[i] = pkg
+		}
+	}
+
 	opts := bundler.Options{
 		Bundle:    &b.bundle,
 		Output:    b.cfg.CreateOpts.Output,
 		TmpDstDir: b.tmp,
 		SourceDir: b.cfg.CreateOpts.SourceDirectory,
 	}
 	bundlerClient := bundler.NewBundler(&opts)
+
 	return bundlerClient.Create()
 }
 
+func (b *Bundle) setPackageRef(pkg types.Package) types.Package {
+	if ref, ok := b.cfg.DevDeployOpts.Ref[pkg.Name]; ok {
+		pkg.Ref = ref
+		errMsg := fmt.Sprintf("Unable to access %s:%s", pkg.Repository, pkg.Ref)
+
+		// Get SHA from registry
+		url := fmt.Sprintf("%s:%s", pkg.Repository, pkg.Ref)
+
+		platform := ocispec.Platform{
+			Architecture: config.GetArch(),
+			OS:           oci.MultiOS,
+		}
+		remote, err := zoci.NewRemote(url, platform)
+		if err != nil {
+			message.Fatalf(err, errMsg)
+		}
+		if err := remote.Repo().Reference.ValidateReferenceAsDigest(); err != nil {
+			manifestDesc, err := remote.ResolveRoot(context.TODO())
+			if err != nil {
+				message.Fatalf(err, errMsg)
+			}
+			pkg.Ref = pkg.Ref + "@sha256:" + manifestDesc.Digest.Encoded()
```

Comment:
```suggestion
			pkg.Ref = ref + "@sha256:" + manifestDesc.Digest.Encoded()
```

Should this reference the `ref` passed in by the flag instead of the original `pkg.Ref`?

Reply:
This would also correct the bundle YAML that gets printed to the screen, right now it shows the old ref

Reply:
^ maybe not actually, I keep forgetting that `dev deploy` creates local bundles